crates/wasmparser/src/validator/operators.rs

```diff
@@ -3968,26 +3990,20 @@ where
         self.push_operand(ValType::Ref(RefType::I31))
     }
     fn visit_i31_get_s(&mut self) -> Self::Output {
-        match self.pop_ref()? {
-            Some(ref_type) => match ref_type.heap_type() {
-                HeapType::I31 => self.push_operand(ValType::I32),
-                _ => bail!(self.offset, "ref heap type mismatch: expected i31"),
-            },
-            _ => bail!(self.offset, "type mismatch: expected (ref null? i31)"),
-        }
+        self.pop_operand(Some(ValType::Ref(
+            RefType::new(true, HeapType::I31).unwrap(),
```

Comment:
Could this perhaps be an associated const on `RefType` e.g. `RefType::I31` or similar?